x-pack/plugins/uptime/public/lib/helper/component_test_helpers.tsx

```diff
@@ -0,0 +1,210 @@
+import { ChromeBreadcrumb } from 'kibana/public';
+import { KibanaContextProvider } from '../../../../../../src/plugins/kibana_react/public';
+import { MountWithReduxProvider } from './helper_with_redux';
+import { AppState } from '../../state';
+
+/* default mock core */
+const defaultCore = coreMock.createStart();
+const mockCore: () => any = () => {
+  let breadcrumbObj: ChromeBreadcrumb[] = [];
+  const core = {
+    ...defaultCore,
+    application: {
+      getUrlForApp: () => '/app/uptime',
+      navigateToUrl: jest.fn(),
+    },
+    chrome: {
+      setBreadcrumbs: (newBreadcrumbs: ChromeBreadcrumb[]) => {
+        breadcrumbObj = newBreadcrumbs;
+      },
+    },
+  };
+
+  return core;
+};
+
+/* Higher Order Components */
+export function withKibanaContext<T>(
+  WrappedComponent: React.ComponentType<T>,
+  { kibanaProps, customCoreOptions }: { kibanaProps?: { services: any }; customCoreOptions?: any }
+) {
+  const core = {
+    ...mockCore(),
+    customCoreOptions,
+  };
+  return (props: any) => (
+    <KibanaContextProvider services={{ ...core }} {...kibanaProps}>
+      <I18nProvider>
+        <WrappedComponent {...(props as T)} />
+      </I18nProvider>
+    </KibanaContextProvider>
+  );
+}
+
+export function withRouter<T>(WrappedComponent: React.ComponentType<T>, customHistory: History) {
+  const history = customHistory || createMemoryHistory();
+  return (props: T) => (
+    <Router history={history}>
+      <WrappedComponent {...(props as T)} />
+    </Router>
+  );
+}
+
+/* Mock Provider Components */
+export function MockKibanaProvider({
+  children,
+  customCoreOptions,
+  kibanaProps,
+}: {
+  children: ReactElement;
+  customCoreOptions?: any;
+  kibanaProps?: { services: any };
+}) {
+  const core = {
+    ...mockCore(),
+    customCoreOptions,
+  };
+  return (
+    <KibanaContextProvider services={{ ...core }} {...kibanaProps}>
+      <I18nProvider>{children}</I18nProvider>
+    </KibanaContextProvider>
+  );
+}
+
+export function MockRouter({
+  children,
+  customCoreOptions,
+  customHistory,
+  kibanaProps,
+}: {
+  children: ReactElement;
+  customCoreOptions?: any;
+  customHistory?: History;
+  kibanaProps?: { services: any };
+}) {
+  const history = customHistory || createMemoryHistory();
+  return (
+    <Router history={history}>
+      <MockKibanaProvider customCoreOptions={customCoreOptions} kibanaProps={kibanaProps}>
+        {children}
+      </MockKibanaProvider>
+    </Router>
+  );
+}
+
+/* React testing library custom render functions */
+export const renderTLWithKibana = (
+  ui: ReactElement,
+  {
+    customCoreOptions,
+    kibanaProps,
+    renderOptions,
+  }: {
+    customCoreOptions?: any;
+    kibanaProps?: { services: any };
+    renderOptions?: any;
+  } = {}
+) => {
+  return render(
+    <MockKibanaProvider {...kibanaProps} customCoreOptions={customCoreOptions}>
+      {ui}
+    </MockKibanaProvider>,
+    renderOptions
+  );
+};
+
+export const renderTLWithRouter = (
```

Comment:
my initial though is that, imo, i don't think we should name it like this renderWithRouter or renderWithKibana stuff. 

I wish we can find a better naming? maybe just have it as simple render, and by default we should add kibana, router and redux for all components, since most components in uptime are dependent those. 

I don't think we will have a use case where adding those by default will cause problems.

Reply:
The one reason I didn't name it `renderWithRouter` is because we already have a `renderWithRouter` in this same file that we are using for enzyme. 

I will consider consolidating into one render function.